subsys/bluetooth/mesh/proxy_cli.c

```diff
@@ -183,43 +179,112 @@ static bool has_net_id(struct bt_mesh_subnet *sub, void *user_data)
 		}
 	}
 
-	if (!memcmp(sub->keys[0].net_id, res->net_id, 8) ||
-	    (bt_mesh_subnet_has_new_key(sub) &&
-	     !memcmp(sub->keys[1].net_id, res->net_id, 8))) {
-		res->srv = srv;
+	/* If net_id is NULL we already know that the networks match */
+	if (!net_id || !memcmp(sub->keys[0].net_id, net_id, 8) ||
```

Comment:
You could pass key index to the function, it will simplify `if` a lot.

Reply:
Should it consider key refersh phase?

Reply:
Well, if Lingao was right when he first implemented this I think this is the right implementation. The parsing of priv net id is just mirroring the implementation for net id.